### PR TITLE
Minor p1 issue fixes

### LINKED ISSUE #52: Problem with getElementById on shadowRoot
In a custom element, I'm trying something like this : 

```
this.elementShadow = this.attachShadow({ mode: 'open' });
this.elementShadow.innerHTML = `<div id="my-id"></div>`;
this.elementShadow.getElementById('my-id');
```

It's OK in a browser with native support, but using the polyfill in a browser not supporting ShadowDOM v1 (Firefox for example), it returns `null`.

It works if I do : 
```
this.elementShadow.querySelector('#my-id');
```

### PR DESCRIPTION
**Pull request by sorvell**
Fixes #52, #30, #119, #96, #133, #134

### CODE REVIEW

File: src/patch-builtins.js [hunk 1]
```diff
@@ -177,6 +179,11 @@ let elementMixin = utils.extendAll({
    */
   get assignedSlot() {
     return getAssignedSlot(this);
+  },
+
+  dispatchEvent(event) {
```
Comment (kevinpschaaf) [inline src/patch-builtins.js]: Technically, `dispatchEvent` also exists on e.g. text nodes, not just elements.  Rather cornery though.

File: tests/shady-dynamic.html [hunk 2]
```diff
@@ -629,11 +639,47 @@
     assert.equal(projected.parentNode, rere);
   });
 
+  test('parentElement', function() {
+    var rere = testElement.shadowRoot.querySelector('x-rereproject');
+    var projected = testElement.shadowRoot.querySelector('#projected');
+    assert.equal(testElement.parentElement, wrap(document.body));
```
Comment (kevinpschaaf) [inline tests/shady-dynamic.html]: What is `wrap` for?

File: src/patch-builtins.js (hunk 1 above)
Comment (sorvell) [inline src/patch-builtins.js]: good catch, should be on Node.

File: tests/shady-dynamic.html (hunk 2 above)
Comment (sorvell) [inline tests/shady-dynamic.html]: legacy, will remove.

**Pull request merged**